### DISCUSSION TITLE
Installing xontribs with distro package?

### DISCUSSION THREAD

**Discussion by Supreeeme** [xonsh/xonsh#5723] [Q&A]
I have xonsh installed through apt, and everything works all fine and good, but I wanted to set up a virtualenv. However, trying to install xontrib-vox as suggested in the tutorial gives me this:
```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.12/README.venv for more information.
```

Is this intended? Am I supposed to be able to install xontribs when using a distro package?

**Comment by anki-code** [comment]
Installing xonsh using system installers like apt is not recommended for the long life systems. The detailed explanation is [here](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#install-xonsh).
The error you see just tells you "You're going to use the approach that close to the system.". For example if you will run update the system in the future and Python version will be updated you can break things. So this is why I recommend to use something isolated like [xonsh-install](https://github.com/anki-code/xonsh-install).
If the system is in container or short life feel free to force the installation.
